components/blitz/src/omero/gateway/facility/TablesFacility.java

```diff
@@ -0,0 +1,526 @@
+                    MaskData[] rowData = new MaskData[nRows];
+                    for (int j = 0; j < nRows; j++) {
+                        MaskData md = new MaskData(mc.x[j], mc.y[j], mc.w[j],
+                                mc.h[j], mc.bytes[j]);
+                        rowData[j] = md;
+                    }
+                    dataArray[i] = rowData;
+                    types[i] = MaskData.class;
+                }
+                if (col instanceof PlateColumn) {
+                    PlateData[] rowData = new PlateData[nRows];
+                    long tableData[] = ((PlateColumn) col).values;
+                    for (int j = 0; j < nRows; j++) {
+                        Plate p = new PlateI(tableData[j], false);
+                        rowData[j] = new PlateData(p);
+                    }
+                    dataArray[i] = rowData;
+                    types[i] = PlateData.class;
+                }
+                if (col instanceof RoiColumn) {
+                    ROIData[] rowData = new ROIData[nRows];
+                    long tableData[] = ((RoiColumn) col).values;
+                    for (int j = 0; j < nRows; j++) {
+                        Roi p = new RoiI(tableData[j], false);
+                        rowData[j] = new ROIData(p);
+                    }
+                    dataArray[i] = rowData;
+                    types[i] = ROIData.class;
+                }
+                if (col instanceof StringColumn) {
+                    String[] rowData = ((StringColumn) col).values;
+                    dataArray[i] = rowData;
+                    types[i] = String.class;
+                }
+                if (col instanceof WellColumn) {
+                    WellSampleData[] rowData = new WellSampleData[nRows];
+                    long tableData[] = ((WellColumn) col).values;
+                    for (int j = 0; j < nRows; j++) {
+                        WellSample p = new WellSampleI(tableData[j], false);
+                        rowData[j] = new WellSampleData(p);
+                    }
+                    dataArray[i] = rowData;
+                    types[i] = ROIData.class;
+                }
+            }
+            return new TableData(header, descriptions, types, dataArray);
+        } catch (Exception e) {
+            handleException(this, e, "Could not load table data");
+        }
+        return null;
+    }
+
+    /**
+     * Get all available tables for a the specified object
+     * 
+     * @param ctx
+     *            The {@link SecurityContext}
+     * @param parent
+     *            The {@link DataObject}
+     * @return See above
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public Collection<FileData> getAvailableTables(SecurityContext ctx,
```

Comment:
@manics recently adding something similar to the CLI -- it might be time to move that query to the server.